<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DetectorSettings</class>
 <widget class="QDialog" name="DetectorSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>500</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>450</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>탐지 설정</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>10</number>
   </property>
   <item>
    <widget class="QGroupBox" name="detectorTypeGroup">
     <property name="title">
      <string>탐지 엔진 선택</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QRadioButton" name="opencvRadio">
        <property name="text">
         <string>OpenCV</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="yoloRadio">
        <property name="text">
         <string>YOLO</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="deepstreamRadio">
        <property name="text">
         <string>DeepStream</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QStackedWidget" name="settingsStack">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="opencvPage">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="opencvModelGroup">
         <property name="title">
          <string>OpenCV 모델</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QComboBox" name="opencvModelCombo">
            <item>
             <property name="text">
              <string>HOG-SVM</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Haar Cascade</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>DNN 기반 모델</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <layout class="QFormLayout" name="formLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>스케일 팩터:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="opencvScaleSpinBox">
              <property name="decimals">
               <number>2</number>
              </property>
              <property name="minimum">
               <double>1.010000000000000</double>
              </property>
              <property name="maximum">
               <double>2.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.010000000000000</double>
              </property>
              <property name="value">
               <double>1.050000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>최소 이웃:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="opencvMinNeighborsSpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
              <property name="value">
               <number>3</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="yoloPage">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="yoloVersionGroup">
         <property name="title">
          <string>YOLO 버전</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QComboBox" name="yoloVersionCombo">
            <item>
             <property name="text">
              <string>v4</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>v5</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>v7</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>v8</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="yoloModelGroup">
         <property name="title">
          <string>모델</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QComboBox" name="yoloModelCombo">
            <item>
             <property name="text">
              <string>YOLOv4-tiny</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>YOLOv4</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="yoloParamsGroup">
         <property name="title">
          <string>파라미터 설정</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>신뢰도 임계값:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QSlider" name="yoloConfidenceSlider">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>99</number>
                </property>
                <property name="value">
                 <number>40</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="yoloConfidenceLabel">
                <property name="minimumSize">
                 <size>
                  <width>40</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>0.40</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>NMS 임계값:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QSlider" name="yoloNmsSlider">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>99</number>
                </property>
                <property name="value">
                 <number>40</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="yoloNmsLabel">
                <property name="minimumSize">
                 <size>
                  <width>40</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>0.40</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>입력 크기:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="yoloInputSizeCombo">
              <item>
               <property name="text">
                <string>320x320</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>416x416</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>512x512</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>608x608</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="objectSelectionButton">
         <property name="text">
          <string>탐지 객체 선택...</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="deepstreamPage">
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="deepstreamModelGroup">
         <property name="title">
          <string>DeepStream 모델</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QComboBox" name="deepstreamModelCombo">
            <item>
             <property name="text">
              <string>ResNet-10</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>ResNet-18</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>ResNet-34</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="deepstreamParamsGroup">
         <property name="title">
          <string>파라미터 설정</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_10">
          <item>
           <layout class="QFormLayout" name="formLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>신뢰도 임계값:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QSlider" name="deepstreamConfidenceSlider">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>99</number>
                </property>
                <property name="value">
                 <number>40</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="deepstreamConfidenceLabel">
                <property name="minimumSize">
                 <size>
                  <width>40</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="text">
                 <string>0.40</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>배치 크기:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="deepstreamBatchSizeSpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>8</number>
              </property>
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="applyButton">
       <property name="text">
        <string>적용</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>취소</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>